### ISSUE TITLE
Can't Discover PC After Some Time

### ISSUE THREAD

**Issue by XiXAleXiX** [mariotaku/moonlight-tv#218]
Hello and once again, thank you for all your work :)

Ok so I am going to describe how I got here step by step:

- I had installed moonlight on my LG TV through webOS Dev manager.

- I installed moonlight on my pc and adjusted all the relevant settings and tl:dr the moonlight app on the tv automatically saw my pc and successfully connected.

- The next day I opened the moonlight app on my pc and the moonlight app on the tv but the tv side was not locating the pc. I tried manually adding the ip of the pc and still nothing. On the contrary, the moonlight app on the tv was getting stuck everytime I tried the connection and I had to restart the tv.

- I uninstalled both apps, reinstalled them, same issue. 

- Couple of days of trying later, I found out about ihsplay and installed it on the tv through webOS Dev manager **and it works (repeatedly).**

- Today I found out about sunshine, I installed it on my pc, followed "MikeTheTech"'s instructions, turned UPnP on but sadly the moonlight app on the tv still does not see the pc.

- I installed moonlight on my iPhone and again it cannot see my pc.

- I have checked (with my amateur eye) the firewall settings of my pc and everything appears to be in order.

Any ideas? What's weird to me (besides it working the first time) is that ihsplay works like a charm and that webOS Dev manager also sees the tv and installs/uninstalls/opens apps normally....

**Comment by mariotaku** [comment]
Is it possible that your computer has changed its IP address?

**Comment by XiXAleXiX** [comment]
No it hasn’t, and even if it did, when I am inputting the ip to add the pc manually I see it from my router’s page :/

**Comment by mariotaku** [comment]
Since you're not able to discover by other clients as well, there's a chance that the ports are not properly opened. Are you able to open sunshine settings webpage on your computer?

**Comment by XiXAleXiX** [comment]
Yes, sunshine is opening normally. I have enabled UPnP through the settings.

**Comment by Shirk** [comment]
> - I installed moonlight on my iPhone and again it cannot see my pc.

This single point pretty much rules out that Moonlight-TV is the source of the problem because Moonlight on iOS is an entirely different app (moonlight-common core aside) and shows the same behaviour. 

UPnP won't do anything for you since you're trying to connect from a device in your local network and not from the outside internet (UPnP opens ports on your router so devices from the internet can connect, it has no effect otherwise).

My best guess would be that your windows firewall might be the issue by blocking the port that sunshine listens to.
This would not prevent you from opening the sunshine control center in your browser _on your PC_ but it would prevent anything else connecting to sunshine. You could try opening the configured port in your windows firewall settings - the default is 47989. At the same time it would be a good idea to check if your firewall profile is set to 'private' for your LAN connection, Windows does disable most discovery services and port access in public network mode.

**Comment by XiXAleXiX** [comment]
Thank you very much for your response. Since posting my issue I have tried streaming with sunshine and I had the same problem. I have then cleanly reinstalled windows because I had issues with windows update, so hopefully everything will work smoothly now. I think though that I will wait for the next release of sunshine because I experienced a bug while trying to connect. Correct me if I am wrong but streaming through moonlight has an expiration date due to Nvidia removing gamestream. Since this is moonlight's issues page and I will proceed with sunshine, maybe this thread should be closed. If not, I will update here.
